libs/common/RespLengthEncodingUtils.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+
+namespace Garnet.common;
+
+/// <summary>
+/// Utils for working with RESP length encoding
+/// </summary>
+public static class RespLengthEncodingUtils
+{
+    /// <summary>
+    /// Maximum length that can be encoded
+    /// </summary>
+    private const int MaxLength = 0xFFFFFF;
+
+    /// <summary>
+    /// Try read RESP-encoded length
+    /// </summary>
+    /// <param name="input"></param>
+    /// <param name="length"></param>
+    /// <param name="bytesRead"></param>
+    /// <returns></returns>
+    public static bool TryReadLength(ReadOnlySpan<byte> input, out int length, out int bytesRead)
+    {
+        length = 0;
+        bytesRead = 0;
+        if (input.Length < 1)
+        {
+            return false;
+        }
+
+        var firstByte = input[0];
+        switch (firstByte >> 6)
+        {
+            case 0:
+                bytesRead = 1;
+                length = firstByte & 0x3F;
+                return true;
+            case 1 when input.Length > 2:
```

Comment:
I had small mistake in my snippet, sorry! 😅 
```suggestion
            case 1 when input.Length > 1:
```

Reply:
done